### ISSUE TITLE
Collection changed callback does not respect input or output as device type

### ISSUE THREAD

**Issue by achronop** [djg/audioipc-2#50]
Registering a callback for input or output collection change is supported by Linux backends (both C and rust). The following code should be supported:
```
  int rv = cubeb_register_device_collection_changed(GetCubebContext(),
       static_cast<cubeb_device_type>(CUBEB_DEVICE_TYPE_INPUT | CUBEB_DEVICE_TYPE_OUTPUT),
       &mozilla::CubebDeviceEnumerator::AudioDeviceListChanged_s, this);
```
With cubeb remote on, the collection change callback is not triggered on plug or unplug. However, the method returns CUBEB_OK and does not give the opportunity to the client to workaround the missing callback.

**Comment by kinetiknz** [comment]
Note to self: `cubeb_stream_register_device_changed_callback` is also unimplemented.

**Comment by kinetiknz** [comment]
Note to self: this needs to be based on the changes made in BMO 1498242

**Comment by kinetiknz** [comment]
https://github.com/djg/audioipc-2/commit/9600391d94854d971ed723e3499c0ab8ede3045b merged Alex's patch that at least fixed the wrong error code.  Actual implementation of the callbacks coming soon.

**Comment by kinetiknz** [comment]
Closed via #62.
